### PR TITLE
feat(sheet): styles on rows and columns

### PR DESCRIPTION
**Pull request by VicKun4937**
document   Pull request: https://github.com/dream-num/univer.ai/pull/332

- support default style in work sheet
 facade API:
```javascript
const fwb = univerAPI.getActiveWorkbook()
const fws = fwb.getActiveSheet()
// set default style
fws.setDefaultStyle('uJSelZ')
// set default column style
fws.setColumnDefaultStyle(3,'uJSelZ')
fws.setColumnDefaultStyle(3,{bg:{rgb:'red'}})
// reset the column default style
fws.setColumnDefaultStyle(3,undefined)

// set row default style
fws.setRowDefaultStyle(1,'uJSelZ')
```

Using in workbook Data:

```typescript
 interface IWorksheetData{
    // other
    /**
     * @property {string|Nullable<IStyleData>} [defaultStyle] - Default style id or style data of Worksheet.
     */
    defaultStyle?: Nullable<IStyleData> | string;

}

interface IRowData{
    // other
    /**
     * style id
     */
    s?: Nullable<IStyleData | string>;
}

interface IColumnData{
     // other
     /**
     * style id
     */
    s?: Nullable<IStyleData | string>;
}
```

### CODE REVIEW

File: packages/sheets/src/commands/mutations/set-worksheet-default-style.mutations.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    subUnitId: string;
+    defaultStyle: string | IStyleData;
+}
+
+export const SetWorksheetDefaultStyleMutation: IMutation<ISetWorksheetDefaultStyleMutationParams> = {
+    id: 'sheet.mutation.set-worksheet-default-style',
+    type: CommandType.MUTATION,
+    handler: (accessor, params) => {
+        const { unitId, defaultStyle, subUnitId } = params;
+        const universheet = accessor.get(IUniverInstanceService).getUniverSheetInstance(unitId);
```
Comment (wzhudev) [inline packages/sheets/src/commands/mutations/set-worksheet-default-style.mutations.ts:32]: Use `getSheetCommandTarget` instead.

File: packages/sheets/src/commands/mutations/set-worksheet-row-column-style.mutation.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+export interface ISetWorksheetRowColumnStyleMutationParams {
+    unitId: string;
+    subUnitId: string;
+    isRow: boolean;
+    styles: ISetWorksheetRowColumnStyleMutationParamsUnit[];
+}
+
+export const SetWorksheetRowColumnStyleMutationFactory = (accessor: IAccessor, params: ISetWorksheetRowColumnStyleMutationParams) => {
+    const universheet = accessor.get(IUniverInstanceService).getUniverSheetInstance(params.unitId);
```
Comment (wzhudev) [inline packages/sheets/src/commands/mutations/set-worksheet-row-column-style.mutation.ts:33]: Same above.

File: packages/sheets/src/services/sheet-interceptor/sheet-interceptor.service.ts [hunk 3]
```diff
@@ -229,6 +231,7 @@ export class SheetInterceptorService extends Disposable {
     private _interceptWorkbook(workbook: Workbook): void {
         const disposables = new DisposableCollection();
         const unitId = workbook.getUnitId();
+        // const isRowStylePrecedeColumnStyle = this._isRowStylePrecedeColumnStyle;
```
Comment (wzhudev) [inline packages/sheets/src/services/sheet-interceptor/sheet-interceptor.service.ts:234]: Remove these commented code.

**Review by wzhudev — CHANGES_REQUESTED** [review]
Overall LGTM, though some code readability issues need to addressed.

File: packages/core/src/shared/tools.ts [hunk 4]
```diff
@@ -690,6 +692,18 @@ export class Tools {
...
+
+      /**
+       * @static
+       * @param {unknown} object Modify the property while leaving the reference unchanged.
+       * @param {unknown} source The source  being merged in object.
+       * @param {(value: unknown, originValue: unknown, key: string, object: unknown, source: unknown, stack: string[]) => {}} [customize]
+       * @return {*}
+       * @memberof Tools
+       */
+    static mergeWith(object: unknown, source: unknown, customize?: (value: unknown, originValue: unknown, key: string, object: unknown, source: unknown, stack: string[]) => {}) {
```
[the remark this answers is not in the document]
Comment (VicKun4937) [reply packages/core/src/shared/tools.ts:704]: removed

File: packages/engine-render/src/components/sheets/sheet-skeleton.ts [hunk 5]
```diff
@@ -1885,7 +1886,15 @@ export class SpreadsheetSkeleton extends Skeleton {
...
         const cell = this.worksheet.getCell(row, col) || this.worksheet.getCellRaw(row, col);
-        const style = this._styles.getStyleByCell(cell);
+        // const style = this._styles.getStyleByCell(cell);
+
+        const cellStyle = this._styles.getStyleByCell(cell);
+        const columnStyle = this.worksheet.getColumnStyle(col) as IStyleData;
+        const rowStyle = this.worksheet.getRowStyle(row) as IStyleData;
+        const defaultStyle = this.worksheet.getDefaultCellStyleInternal();
+
+        const style = false ? composeStyles(defaultStyle, columnStyle, rowStyle, cellStyle) : composeStyles(defaultStyle, rowStyle, columnStyle, cellStyle);
```
[the remark this answers is not in the document]
Comment (VicKun4937) [reply packages/engine-render/src/components/sheets/sheet-skeleton.ts:1896]: code error, it should be from config serivce, _isRowStylePrecedeColumnStyle

**Review by wzhudev — APPROVED** [review]
LGTM

**Pull request merged**